### DISCUSSION TITLE
Does MSW work with fetch server-side?

### DISCUSSION THREAD

**Discussion by thany** [mswjs/msw#1641] [Q&A]
I couldn't get it to work, I had to migrate Node.js native `fetch` over to `axios` which feels like a waste.

What are the steps to be able to use `fetch` during server-side execution please?

**Comment by kettanaito** [comment]
Hi!

MSW works with both native fetch and third-party libraries like Axios in Node.js. Was there a particular issue you experience that forced you to migrate? Share it below, chances are I've seen it before and can help you.

This is also a nice time to encourage you to migrate to msw@next that utilizes the Fetch API primitive as first class citizens in both browser and Node.js. Read more about it in https://github.com/mswjs/msw/discussions/1464.

**Reply by thany** [reply]
Well yeah, we used `fetch` (on the server, still) to do API requests to a remote server. Those requests were just passing through. The *only* thing we changed, is to use axios to make the very same requests, and they would be caught by MSW perfectly fine, exactly as intended.

Unfortunately, we *had* to move on and go with axios. And I don't see our project migrating back to fetch anytime soon, partly because we ran into this.

But if you're saying it ought to work, then I'm sure we just ran into a 100% reproducible bug for every request that only surfaced when using `fetch`. And since we already moved on, there's no way to dive into it anymore 😔

**Reply by kettanaito** [reply]
Got it. If you ever catch that bug again, please make sure to submit it. Thanks!
